pandas/core/indexes/base.py

```diff
@@ -2404,6 +2407,29 @@ def is_all_dates(self) -> bool:
         )
         return self._is_all_dates
 
+    @final
+    @classmethod
+    def _is_numeric_index(cls) -> bool:
+        """
+        Check if this is a NumericIndex, but *not* Int64Index, UInt64Index, FloatIndex.
+
+        Used to check if an operation should return NumericIndex or plain Index.
+        """
```

Comment:
this should just be a property on the index itself no? why is this needed at all

Reply:
Yes, I've changed it to an attribute on the index classes.